test/jdk/javax/swing/JMenu/4213634/bug4213634.java

```diff
@@ -77,33 +71,30 @@ public  void createAndShowGUI() {
         frame.getContentPane().add("Center", ta);
         JButton button = new JButton("Test");
         frame.getContentPane().add("South", button);
-        frame.setBounds(100, 100, 400, 400);
+        frame.setSize(400, 400);
+        frame.setLocationRelativeTo(null);
         frame.setVisible(true);
         button.requestFocusInWindow();
     }
 
-    private void test() throws AWTException, InterruptedException, InvocationTargetException {
-        Robot robot = new Robot();
-        robot.setAutoDelay(50);
-        robot.waitForIdle();
+    private static void test() throws AWTException, InterruptedException, InvocationTargetException {
 
         Util.hitMnemonics(robot, KeyEvent.VK_1);
         robot.waitForIdle();
+        robot.delay(100);
 
-        SwingUtilities.invokeAndWait(new Runnable() {
-            @Override
-            public void run() {
-                if (!menu.isSelected()) {
-                    throw new RuntimeException(
-                        "Failed: Menu didn't remain posted at end of test");
-                } else {
-                    System.out.println("Test passed!");
-                    frame.dispose();
-                }
+        SwingUtilities.invokeAndWait(() -> {
+            frame.dispose();
+            if (!menu.isSelected()) {
```

Comment:
Previously, the condition `!menu.isSelected()` was verified before the frame was disposed of.

I think you should verify whether menu is select before disposing of the frame; once the frame is disposed of, the state of components is not well-defined.

Reply:
I agree, frame.dispose() can be added within finally block instead of here.